compiler/src/scala/quoted/runtime/impl/QuotesImpl.scala

```diff
@@ -1406,6 +1406,45 @@ class QuotesImpl private (using val ctx: Context) extends Quotes, QuoteUnpickler
       end extension
     end TypeCaseDefMethods
 
+
+    type Wildcard = tpd.Ident
+
+    object WildcardTypeTest extends TypeTest[Tree, Wildcard]:
+      def unapply(x: Tree): Option[Wildcard & x.type] = x match
+        case x: (tpd.Ident & x.type) if x.name == nme.WILDCARD => Some(x)
+        case _ => None
+    end WildcardTypeTest
+
+    object Wildcard extends WildcardModule:
+      def apply(): Wildcard =
+        withDefaultPos(untpd.Ident(nme.WILDCARD).withType(dotc.core.Symbols.defn.AnyType))
+      def unapply(pattern: Wildcard): true = true
+    end Wildcard
+
+    type TypedTree = tpd.Typed
+
+    object TypedTreeTypeTest extends TypeTest[Tree, TypedTree]:
+      def unapply(x: Tree): Option[TypedTree & x.type] = x match
+        case x: (tpd.Typed & x.type) => Some(x)
+        case _ => None
+    end TypedTreeTypeTest
+
+    object TypedTree extends TypedTreeModule:
+      def apply(expr: Term, tpt: TypeTree): Typed =
+        withDefaultPos(tpd.Typed(xCheckMacroValidExpr(expr), tpt))
+      def copy(original: Tree)(expr: Term, tpt: TypeTree): Typed =
+        tpd.cpy.Typed(original)(xCheckMacroValidExpr(expr), tpt)
+      def unapply(x: Typed): (Term, TypeTree) =
+        (x.expr, x.tpt)
+    end TypedTree
+
+    given TypedTreeMethods: TypedTreeMethods with
+      extension (self: Typed)
+        def tree: Tree = self.expr
+        def tpt: TypeTree = self.tpt
+      end extension
```

Comment:
mere style question whether the "double end" is desirable? Arguably, both ends on a very short definition? It comes up on the forum and folks look at this source for style guidance. Maybe we need an influencer on twitter to offer style advice.

Reply:
Not sure about the general desirability of the "double end". It mostly came from copy-pasting other uses. I used regex find and replace in this file a lot, the explicit end is sometimes useful for that.

I would say the "double end" is useful when we have a large number of extension methods. In that case it makes it easier to see that all the methods are extension methods.